### PR TITLE
Update journal.rs

### PR DESCRIPTION
**Pull request by kteza1**
I think you should use `splitn(2..` for 1 split

http://is.gd/BYkUt4

### CODE REVIEW

File: src/journal.rs [hunk 1]
```diff
@@ -128,7 +128,7 @@ impl Journal {
             unsafe {
                 let b = ::std::slice::from_raw_parts_mut(data, sz as usize);
                 let field = ::std::str::from_utf8_unchecked(b);
-                let mut name_value = field.splitn(1, '=');
```
Comment (jmesmon) [inline src/journal.rs]: Yep, not sure how this ever worked.

File: src/journal.rs [hunk 2]
```diff
@@ -66,7 +60,8 @@ pub type JournalRecord = BTreeMap<String, String>;
 ///
 /// Supports read, next, previous, and seek operations.
 pub struct Journal {
-    j: ffi::sd_journal
+    j: ffi::sd_journal,
+    cursor: String,
```
Comment (jmesmon) [inline src/journal.rs]: Why do we need to cache a copy of the cursor here? Why not just have an accessor that retrieves it via ffi?

Comment (kteza1) [inline src/journal.rs]: I'm using this for iterator. I'll create a function (wrapper) which gets current cursor and use it if you prefer it that way

**Comment by jmesmon** [comment]
- It looks like there are a bunch of formatting changes mixed in here, it'd be preferable if they were a seperate commit, and how they were generated was noted in the commit (I'm presuming they are automated).
- The commit message for the first commit needs improvement.
- The test that was added should be enabled
- There a a bunch of unused variables that should be removed
- The change to add the iterator should be a single commit (not split into 2)
- Name accessors `noun`, not `get_noun` (wrt `get_cursor`).
- Instead of copying the cursor a few times, use a `CString::from_raw` to take ownership
- When iterating over a `Journal` is the current cursor always needed? Will we at some point have folks looking for iteration without that extra information (allocating & copying it isn't free, presumably). What is the long term story for handling both needing and not needing the cursor value while iterating?

**Comment by kteza1** [comment]
* Formatting changes are automated by rustfmt (with sublime plugin). Will send it in a separate commit (If its ok. Pls let me know)
* Didn't enable the test because it's blocking and test won't complete
* I'll remove unnecessary uses in tests but couldn't get rid of `wait_time` as rust is not recognizing uses in unsafe code (tested on 1.5. Will try with 1.7)
* If you don't want this library to not have any unnecessary overhead, I think it's better I remove cursor from the iterator. Users can always use get_cursor (will rename this) to get it while they are iterating. Let me know if there is a better idea :)

And shall I close this and open a new PR?

**Comment by jmesmon** [comment]
- On formatting changes: "it'd be preferable if they were a seperate commit, and how they were generated was noted in the commit". So, yep. Separate commit is fine.

- On the test being blocking: perhaps this indicates that the timeout should be configurable? Or maybe we could just use `.take(10)` or so to limit how many journal elements are obtained? I suppose that in addition we might need to emit some journal messages to ensure there are at least `N` messages in a reasonable time to iterate over.

- On overhead: I don't want to force users to have more overhead than they want, but at the same time I want to be able to allow those that do need those extra items to obtain them without much additional work. I'm not sure how convenient it would be for users to obtain the cursor while iterating. Is there an easy way to allow users to obtain the cursor if the iterator doesn't return it? For the journal, it might be convenient to provide a iteration wrapper or 2 that provide the extra data (cursor) that some users need.

- On new PR: I don't think it's necessary. If you want to have the 1st commit (`split_n`) in a separate PR I'd be happy to merge that immediately.

**Comment by kteza1** [comment]
> instead of copying the cursor a few times, use a CString::from_raw to take ownership

Could you please point where I should do this?

File: src/journal.rs [hunk 3]
```diff
@@ -137,6 +131,63 @@ impl Journal {
...
     }
+
+    pub fn get_cursor(&self) -> Result<String> {
+        // update cursor
+        let mut c_cursor: *mut c_char = ptr::null_mut();
+        let mut cursor: String = "".to_string();
+        if sd_try!(ffi::sd_journal_get_cursor(self.j, &mut c_cursor)) == 0 {
+            unsafe {
+                // You should use Cstr for memory allocated by C
+                cursor = CStr::from_ptr(c_cursor as *const _)
```
Comment (jmesmon) [inline src/journal.rs]: Use `CString::from_raw()` here instead, removing the manual memory management.

**Pull request closed without merging**